bodhi/tests/server/consumers/test_automatic_updates.py

```diff
@@ -89,6 +89,41 @@ def test_consume(self, caplog):
 
         assert not any(r.levelno >= logging.WARNING for r in caplog.records)
 
+    def test_consume_with_orphan_build(self, caplog):
+        """
+        Assert existing builds without an update can be handled.
+
+        Such builds can exist e.g. if they're used in a buildroot override.
+        """
+        caplog.set_level(logging.DEBUG)
+
+        # Run the handler to create the build & update, then remove the update.
+        self.handler(self.sample_message)
+        build = self.db.query(Build).filter_by(nvr=self.sample_nvr).one()
+        update = build.update
+        build.update = None  # satisfy foreign key constraint
+        self.db.delete(update)
+
+        # Now test with the same message again which should encounter the
+        # build already existing in the database.
+        self.handler(self.sample_message)
+
+        # check if the update exists...
+        update = self.db.query(Update).filter(
+            Update.builds.any(Build.nvr == self.sample_nvr)
+        ).first()
+
+        # ...and some of its properties
+        assert update is not None
+        assert update.type == UpdateType.unspecified
+        assert update.status == UpdateStatus.testing
+        assert update.test_gating_status is None
```

Comment:
Shouldn't that status be `waiting`?

Reply:
Valid objection. I guess why the tests succeed (for me at least) is because we don't actually ask Greenwave in tests, i.e. don't update anything. But we should mock this part, too. I'll get on it.

Reply:
Then again, no: it's because in these tests, test gating isn't required so the field won't be set. All is well. :smiley: